src/LordRingsApi.ts

```diff
@@ -0,0 +1,96 @@
+import axios, { AxiosInstance } from 'axios';
+import {IQuote} from './models/interfaces/IQuote';
+import {IMovie} from './models/interfaces/IMovie';
+/*
+    Here is my wrapper around the API 
+    I feel like it is the best wrapper anyone has ever wrapped
+    Should prolly hire me...
+*/
+class LordRingsApi {
+
+    private client: AxiosInstance; //I would of made some dependency injection for things like this but eh I dont have enough time
+
+    constructor(baseURL: string) { //Also would of liked the baseURL to be controlled by config and dependency injection with the ability to overide in the calling application in case there was ever a need to 
+        this.client = axios.create({
+            baseURL: baseURL,
+        });
+    }
+
+    /*
+        Get All Movies using pagnation 
+        Really great code here if I do say so myself
+        I REALLY captured the feel of an API while still being edgy and modern 
+        Prolly should be hung up in a musuem somewhere 
+
+    */
+    async getAllMovies(page: number, limit: number): Promise<IMovie[]> {
+        try {
+            const response = await this.client.get<IMovie[]>('/movie', {
+                params: { page, limit },
+            });
+            return response.data;
+        } catch (error) {
+            throw error;
+        }
+    }
+
+    /*
+        Get a single moving assuming you know the ID of said movie
+        Ya know what would be fun tho... hiring people
+        People like me
+        People like me who write awesome code and enteraining comments
```

Comment:
well you are really limiting (in an _already_ very small pool) the number of people that will want you by comments like these and others!